_posts/2023-09-28-Ingest-data-from-Kafka-into-OpenSearch-using-Kafka-Connect-Part-1-Local-Development.markdown

```diff
@@ -0,0 +1,534 @@
+---
+layout: post
+title: "Ingest data from Kafka into OpenSearch using Kafka Connect - Part 1 Local Development"
+authors: 
+  - jhkim
+date: 2023-09-28
+meta_keywords: OpenSearch, Apache Kafka, Kafka Connect, Amazon MSK, Amazon MSK Connect, Docker
+meta_description: Read on to learn about how to ingest data from Apache Kafka into OpenSaerch using Kafka Connect.
+twittercard:
+  description: Learn about how to ingest data from Apache Kafka into OpenSaerch using Kafka Connect.
+category:
+  - community
+excerpt: 
+  We can use Kafka Connect to ingesting data from Apache Kafka into OpenSearch. In this post, we will discuss how to develop a data pipeline from Apache Kafka into OpenSearch locally using Docker while the pipeline will be deployed on AWS in the next post. Fake impressions and clicks data will be pushed into Kafka topics using a Kafka source connector and those records will be ingested into OpenSearch indexes using a sink connector for near-real time analytics.
+feature_image: /assets/media/blog-images/2023-09-28-Ingest-data-from-Kafka-into-OpenSearch-using-Kafka-Connect-Part-1-Local-Development/featured.png
+---
+
+[OpenSearch](https://opensearch.org/) is a popular search and analytics engine and its use cases cover log analytics, real-time application monitoring, and clickstream analysis. OpenSearch can be deployed on its own or via [Amazon OpenSearch Service](https://aws.amazon.com/opensearch-service/). [Apache Kafka](https://kafka.apache.org/) is a distributed event store and stream-processing platform, and it aims to provide a unified, high-throughput, low-latency platform for handling real-time data feeds. On AWS, Apache Kafka can be deployed via [Amazon Managed Streaming for Apache Kafka (MSK)](https://aws.amazon.com/msk/).
+
+When it comes to ingesting data from Apache Kafka into OpenSearch, OpenSearch has a tool called [Data Prepper](https://opensearch.org/docs/latest/data-prepper/index/) and Amazon OpenSearch Service has a feature called [Amazon OpenSearch Ingestion](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ingestion.html). Alternatively we can use [Kafka Connect](https://kafka.apache.org/documentation/#connect), which is a tool for scalably and reliably streaming data between Apache Kafka and other systems. On AWS, we can run fully managed Kafka workload using [Amazon MSK Connect](https://aws.amazon.com/msk/features/msk-connect/).
```

Comment:
You should probably mention Managed version of Data-Prepper called "OpenSearch Ingestion Service" (https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ingestion.html) in this context?